src/lib/homestore_backend/heap_chunk_selector.cpp

```diff
@@ -130,11 +135,44 @@ csharedChunk HeapChunkSelector::select_specific_chunk(const chunk_num_t chunkID)
     if (vchunk.get_internal_chunk()) {
         auto& avalableBlkCounter = it->second->available_blk_count;
         avalableBlkCounter.fetch_sub(vchunk.available_blks());
+        remove_chunk_from_defrag_heap(vchunk.get_chunk_id());
     }
 
     return vchunk.get_internal_chunk();
 }
 
+// select_chunk_for_GC will only be called when GC is triggered, and will return the chunk with the most
+// defrag blocks
+csharedChunk HeapChunkSelector::select_chunk_for_GC() {
```

Comment:
nit: Suggest to rename to most_defrag_chunk(), so that we don't keep GC knowledge inside heapchunkselector if possible?